components/Users/FoodAddForm.jsx

```diff
@@ -0,0 +1,225 @@
+// component for displaying food entry form and make a post request to add user inputted data to database
+
+import React, { useState } from "react";
+import { Formik, Form, Field } from "formik";
+import { Alert, Button, Paper, Snackbar, Typography } from "@mui/material";
+import styled from "@emotion/styled";
+import CheckCircleIcon from "@mui/icons-material/CheckCircle";
+import { foodYupSchema } from "../../Func_Definitions/foodYupSchema";
+import { motion } from "framer-motion";
+
+const StyledForm = styled(Form)(() => ({
+  width: "600px",
+  margin: "auto",
+  display: "flex",
+  flexDirection: "column",
+  alignItems: "center",
+  padding: "20px",
+}));
+
+const StyledContainer = styled.div((props) => ({
+  display: "flex",
+  width: "70%",
+  height: "60px",
+  flexDirection: "column",
+  alignItems: "center",
+}));
+
+const StyledField = styled(Field)(() => ({
+  width: "100%",
+  height: "45px",
+  border: "1px solid #ccc",
+  borderRadius: "5px",
+}));
+
+const StyledSelectField = styled(Field)(() => ({
+  width: "100%",
+  height: "45px",
+  border: "1px solid #ccc",
+  borderRadius: "5px",
+}));
+
+const FoodAddForm = (props) => {
+  const [snackbar, setSnackbar] = useState(null);
+  const handleCloseSnackbar = () => setSnackbar(null);
+  let MealDescription = props.MealDescription;
+
+  const saveFoodEntry = async (dataWithUpdatedTime) => {
+    let sessionUser = props.sessionUser;
+    let new_data = {
+      id: "",
+      userEmail: sessionUser.email,
+      ...dataWithUpdatedTime,
+    };
+
+    // api call for adding new food entry with optimistic update
+
+    await props.mutateFoodPerEmail(
```

Comment:
destruct function in the start of the file.
`const {mutateFoodPerEmail} = props.`